src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -438,6 +438,7 @@ private async Task SetRunFromPackageAppSetting(Site functionApp, string runFromP
 
         public async Task PublishZipDeploy(Site functionApp, Func<Task<Stream>> zipFileFactory)
         {
+            (var content, var length) = CreateStreamContentZip(await zipFileFactory());
```

Comment:
`content` will throw a `ObjectDisposedException` on retry below. It needs to be in the retry body

Reply:
Oh right right! I will just move `await zipFileFactory()` out of retry then, and call `CreateStreamContentZip` inside the retry.

I did this because we were retrying if creating a zip file failed due to build problems. I don't think we need to do that.

Reply:
this whole action needs to be rewritten.

Reply:
I agree that it has become messy. Ok, I can leave this change out of this PR as it doesn't concern the issue directly. 
Feel free to assign me an issue for the rewrite if you'd like with any specific suggestions you had.